test/cpp/end2end/async_end2end_test.cc

```diff
@@ -807,7 +807,11 @@ class AsyncEnd2endServerTryCancelTest : public AsyncEnd2endTest {
     EXPECT_FALSE(context->IsCancelled());
     context->TryCancel();
     gpr_log(GPR_INFO, "Server called TryCancel()");
-    EXPECT_TRUE(context->IsCancelled());
+    while (!context->IsCancelled()) {
```

Comment:
So, this is worse than I thought. The issue is that the original code is mixing Next with Pluck, which is a huge no-no. IsCanceled ultimately calls Pluck, and in the case of CANCEL_DURING_PROCESSING, this means that there is potentially a simultaneous Next and Pluck. This is absolutely not ok. The solution has to be to make a non-pluck-based interface to IsCanceled. Let's talk about this.